client/src/plugins/update-checks/UpdateChecks.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import React, { PureComponent } from 'react';
+
+import NewVersionInfoView from './NewVersionInfoView';
+
+import Flags, { SERVER_INTERACTION, UPDATES_SERVER_URL } from '../../util/Flags';
+import Metadata from '../../util/Metadata';
+
+class NoopComponent extends PureComponent {
+  render() {
+    return null;
+  }
+}
+
+const PRIVACY_PREFERENCES_CONFIG_KEY = 'editor.privacyPreferences';
+const LATEST_UPDATE_CHECK_INFO_CONFIG_KEY = 'editor.latestUpdateCheckInfo';
+const EDITOR_ID_CONFIG_KEY = 'editor.id';
+const OS_INFO_CONFIG_KEY = 'os.info';
+
+const UPDATE_CHECK_API = '/update-check';
+
+const HOURS_DENOMINATOR = 3600000;
+const HOURS_LIMIT = 24;
+
+export default class UpdateChecks extends PureComponent {
+
+  constructor(props) {
+    super(props);
+
+    if (!Flags.get(SERVER_INTERACTION) || !Flags.get(UPDATES_SERVER_URL)) {
+      return new NoopComponent();
+    }
+
+    this.state = {
+      showModal: false
+    };
+  }
+
+  async componentDidMount() {
+
+    const {
+      config
+    } = this.props;
+
+    const privacyPreferences = await config.get(PRIVACY_PREFERENCES_CONFIG_KEY);
+    if (!privacyPreferences || !privacyPreferences.ENABLE_UPDATE_CHECKS) {
+      this.setState({ checkNotAllowed: true });
+      return;
+    }
+
+    const latestUpdateCheckInfo = await config.get(LATEST_UPDATE_CHECK_INFO_CONFIG_KEY);
+    if (latestUpdateCheckInfo && !this.isTimeExceeded(latestUpdateCheckInfo.latestUpdateTime)) {
+      this.setState({ checkNotNeeded: true });
+      return;
+    }
+
+    this.setState({ isChecking: true });
+
+    const updatesServerUrl = Flags.get(UPDATES_SERVER_URL);
```

Comment:
These two seem to belong to the latest version check, right?

Reply:
Yes, isChecking state means that the request was sent but the answer has not been received yet. We decided that we want to feed the server URL as a flag as well, that's why we read it from Flags.

Reply:
Sorry, I meant the `checkLatestVersion` method.

So instead of placing `this.setState({ isChecking: true });` in the `componentDidMount`, we'd rather place it inside of `checkLatestVersion`.